targets/drive/mobile/main.jsx

```diff
@@ -126,22 +133,91 @@ const isBackgroundServiceParameter = () => {
   return queryDict.backgroundservice
 }
 
-document.addEventListener(
-  'DOMContentLoaded',
-  () => {
-    if (!isBackgroundServiceParameter()) {
-      loadState().then(renderAppWithPersistedState)
+var app = {
+  store: {},
+
+  initialize: function() {
+    this.bindEvents()
+  },
+
+  bindEvents: function() {
+    document.addEventListener(
+      'deviceready',
+      this.onDeviceReady.bind(this),
+      false
+    )
+    document.addEventListener('resume', this.onResume.bind(this), false)
+    document.addEventListener('pause', this.onPause.bind(this), false)
+  },
+
+  startApplication: async function() {
+    const persistedState = await loadState()
+    const cozyURL = persistedState.mobile
+      ? persistedState.mobile.settings.serverUrl
+      : ''
+    configureReporter()
+    const client = initClient(cozyURL)
+    this.store = configureStore(client, persistedState)
+    const store = this.store
+
+    const clientInfos = store.getState().settings.client
+    if (clientInfos) {
+      const isRegistered = await client.isRegistered(clientInfos)
+      if (isRegistered) {
+        startReplication(store.dispatch, store.getState) // don't like to pass `store.dispatch` and `store.getState` as parameters, big coupling
+        initBar(client)
+      } else {
+        console.warn('Your device is no more connected to your server')
+        store.dispatch(revokeClient())
+      }
     }
+
+    useHistoryForTracker(hashHistory)
+    if (store.getState().mobile.settings.analytics)
+      startTracker(store.getState().mobile.settings.serverUrl)
+
+    const root = document.querySelector('[role=application]')
+
+    render(
+      <I18n
+        lang={getLang()}
+        dictRequire={lang => require(`drive/locales/${lang}`)}
+      >
+        <CozyProvider store={store} client={client}>
+          <DriveMobileRouter history={hashHistory} />
+        </CozyProvider>
+      </I18n>,
+      root
+    )
   },
-  false
-)
 
-document.addEventListener(
-  'deviceready',
-  () => {
-    if (isBackgroundServiceParameter()) {
+  onDeviceReady: async function() {
+    if (!isBackgroundServiceParameter()) {
+      this.startApplication()
+    } else {
       startBackgroundService()
     }
+
+    // this.store.dispatch(backupImages())
+    if (navigator && navigator.splashscreen) navigator.splashscreen.hide()
+    registerShareWithCozyDriveIntent(this.store)
```

Comment:
Here `this.store` is not defined because of the asynchronous start of the application: we load state from local storage (indexedDB which is async).